src/hooks/Tokens.ts

```diff
@@ -181,8 +182,16 @@ export function useCurrency(currencyId: string | null | undefined): Currency | n
   const { chainId } = useActiveWeb3React()
   const isETH = currencyId?.toUpperCase() === 'ETH'
   const token = useToken(isETH ? undefined : currencyId)
-  const extendedEther = useMemo(() => (chainId ? ExtendedEther.onChain(chainId) : undefined), [chainId])
+  const extendedEther = useMemo(
```

Comment:
idk if it translates but this block reminds me of `useCurrency`

https://github.com/Uniswap/interface/blob/f59c5f687742d9d50adf18f0756dff3eb6d659b6/src/hooks/Tokens.ts#L180